cmd/controller-manager/app/controller_manager.go

```diff
@@ -228,13 +229,13 @@ func Run(controllerManagerOptions *options.ControllerManagerServer) error {
 		// 	k8sClientBuilder = rootClientBuilder
 		// }
 
-		err := StartControllers(controllerManagerOptions, k8sKubeconfig, serviceCatalogClientBuilder, recorder, stop)
+		err := StartControllers(controllerManagerOptions, k8sKubeconfig, serviceCatalogClientBuilder, recorder, ctx.Done())
 		glog.Fatalf("error running controllers: %v", err)
 		panic("unreachable")
 	}
 
 	if !controllerManagerOptions.LeaderElection.LeaderElect {
-		run(make(<-chan (struct{})))
+		run(context.TODO())
```

Comment:
Apparently this is fine and being used in Kubernetes code itself...
https://golang.org/pkg/context/#TODO
> TODO returns a non-nil, empty Context. Code should use context.TODO when it's unclear which Context to use or it is not yet available (because the surrounding function has not yet been extended to accept a Context parameter). TODO is recognized by static analysis tools that determine whether Contexts are propagated correctly in a program.